### DISCUSSION TITLE
Иногда переключение страниц в WebUI отваливается до полной перезагрузки контроллера

### DISCUSSION THREAD

**Discussion by vortigont** [vortigont/FireLamp_JeeUI#50] [General]
**Симптомы**: Через некоторое продолжительное время после перзагрузки (порядок часов) веб-интерфейс и через приложение native WLED перестают переключаться вкладки - "настройки", "эффекты" и т.п.
При этом работает вкл\выкл лампы на текущей странцие.

После перезагрузки по питанию всё работает как обычно.

Я на это у себя тоже натыкался, похоже на проблему где-то в сетевом стеке или в WiFi, я пока не понял. По сети просто перестают бегать большие блоки данных, хотя мелкие бегают нормально. Насколько часто это проявляется я пока не понял. Требуется больше вводных данных.

СС @BashSer

**Comment by dina16121974** [comment]
Как отловить? Это на конкретной прошивке? Я вертел лампу ,долго на 16х16 и 64х32. Мне такого не попадалось ,бывали тормоза ,но не такие жесткие. Но я на ви фи грешил ,у меня мышка беспроводная срала ,в эфир. Сейчас на проводе ,вообще проблем нет. Похожое поведение было на 128х64 ,но там Вы сами сказали ,проблемы с памятью. Может ,где то в роутере? Я в этом мало понимаю ,там настроек куча. Еще вспомнил ,у меня пару раз было ,что роутер ,менял адреса ,подключенных еспешек ,после добаления новых еспешек или после зависания роутера ,это я не отследил ,почему так я не знаю.
Если забанят на форуме ,диалог буду вести здесь )))

**Comment by vortigont** [comment]
да пока непонятно. Может и роутер виноват, я докопался что проблема где-то в сети. Проявляется случайно на текущем мастере (но и раньше тоже такое наблюдал), в какой-то момент просто перестает переключать странички, хотя мелкие запросы бегают.
Собираем статискику, значит проявляется не у всех.
 ps. да, на форме имеет место быть какое-то лобби

**Comment by dina16121974** [comment]
Ну я не знаю ,можно как то в вэб морде ,сделать ,что бы состояние памяти и ви фи ,как то сигнализировало цветом или значком? Хорошо или беда. Может так отловим?
<details>
  <summary>SpoilerOff-topic</summary>
В личку написали...
Привет.
Гайвер то ли хохол, то ли еврей.
Человек, который знает его лично, пока не отзывается. Он с украины.
Модераторы на форуме хохол, испанец и кто то ещё. Пока не разрулил.
Все проекты он поддерживает только ради денег. Жаль переписка не сохранилась, в которой он откровенничал.
И ему глубоко насрать на людей, которые приходят на этот форум.

А насчёт Бауманки, где он учился, вполне возможно и ему я диплом писал.

Так что не нужно удивляться тому, что происходит.
Им на это всем глубоко насрать.

PS. Я не имею ничего против любой национальности. Сложно назвать национальность в которой у меня бы не было друзей. Просто нужно быть человеком до конца.


Здесь мое мнение. Хоть скрины начинай сохранять ,что бы потом ,доказывать ,что не верблюд)))
</details>

**Comment by vortigont** [comment]
> можно как то в вэб морде ,сделать ,что бы состояние памяти и ви фи

да память уже есть. Вайфай можно добавить, уже лежит в "задумках". Но закавырка конкретно в этом случае в том что соединение на самом деле не рвется и данные продолжают ходить. Т.е. ни контроллер ни морда не знают что часть сообщений где-то попадает в черную дыру. Всё что я выяснил это то дыра в направлении от контроллера в вобморду.
Постараюсь найти роутер который умеет перехватывать вайфай трафик и поглядеть что там.

CC: @vvip-68

**Reply by vvip-68** [reply]
По моим наблюдениям временами AsyncWebServer или сокет внутри где-то наглухо падает и все что передается сокету просто никуда из контроллера не уходит. Причину не знаю. Ты же любитель-копатель-исследователь - может разберешься что там падает.

Я из браузера с периодичностью в несколько секунд по веб-сокету запрашиваю контроллер объём свободной память и uptime (которые все равно отображаются в веб-морде). Когда вот то что выше падает и от контроллера перестают приходить ответы - я говорю, что связь пропала. Своеобразный пинг.

Со временем это может "отвиснуть" и тогда с приходом ответа - типа связь восстановилась. Или перезагрузка страницы в браузере может восстановить соединение. Или перезагрузка контроллера.

Знаю, скажешь криво все это. Но утилитарно. У меня так.

**Comment by BashSer** [comment]
Дополнение по поводу внутренней сети.
Сначала я грешил на то, что Keenetic меняет каналы(настраивается на роутере, помогает стабильнее работать устройствам 2.4ГГц в многоквартирных домах) - установил статичный канал, не помогло. Отмечу, что китайские устройства на чипах ESP и Tuya(перепиленный esp под своим брендом для своего облака и api ?) стабильно работают при динамической смене канала, их более 5 шт. в сети.
Могу сделать отдельную сеть для esp, даже отдельно для лампы с отдельным vlan, но вряд ли проблема в сети между роутером и esp. Предлагаю не развивать вектор в эту сторону

**Comment by vortigont** [comment]
> Когда вот то что выше падает и от контроллера перестают приходить ответы - я говорю, что связь пропала. Своеобразный пинг.

так в том то и фишка что в нашем случае соединение не падет, пинги с апдейтом памяти ходят, можно включить и выключить лампу. Теряются большие пакеты, построения интерфейса, порядка 4к. Раньше я такого не наблюдал (пол года назад), когда появилось не уследил. На форуме никто не жаловался до того как БашЕр пришел, поэтому грешил на свой корявый вайфай. (У меня и ноут от него наглухо отваливается если работать через 5 ГГц., да так что только ребут роутера помогает).
Можно начать перебирать фреймворки, но это муторно. У меня есть альтернативный асинк-сервер с более качественным кодом, може попробую сделать профиль на нем. Приболел что-то... :( как только, так сразу.

**Reply by vvip-68** [reply]
А, ну у меня то больших пакетов не бегает... Если уж отвалилось - и маленькие не ходят...
Может быть - фрагментация памяти, невозможность выделить достаточно большой буфер под большой пакет?

**Comment by dina16121974** [comment]
То есть считаете ,что проблема в кривых есп? У меня с 15 января отпуск ,могу зарядить ,3 еспешки ,на круглосуточную работу. Щас пойду на почту забирать ,ви фи розетку ,ви фи датчик протечки и ви фи реле,взял на поиграться)))

**Reply by vvip-68** [reply]
Вот, кстати, только сегодня попадалась на глаза статься о разновидностях ESP32 (которые даже не S2/S3/C3, а обычные двухъядерные WROOM), которые в основном продаются сейчас на ali. Большинство чипов - признаны устаревшими и давно сняты с производства в том числе из за аппаратных ошибок, которые приводили к каким-то проблемам в том числе и с WiFi... Возможно ноги еще и отсюда растут, не только проблемы с библиотеками / ядром...

**Comment by vortigont** [comment]
> Большинство чипов - признаны устаревшими и давно сняты с производства в том числе из за аппаратных ошибок, которые приводили к каким-то проблемам в том числе и с WiFi...

да, такое тоже есть. У чипов несколько ревизий и в IDF есть куча флагов для сборки под эти баги. Но т.к. ядро ардуино монолитное, там нельзя ничего менять, там включена прорва всех костылей на все случаи жизни.

**Comment by dina16121974** [comment]
Вот еще вспомнил. Когда люди собирали лампы на есп8266 ,попадались кривые экземпляры. В сети виделись ,но подключаться не давались. Где то помогали костыли ,задержки delay по моему так. Мне попался такой кривой камень ,я по моему статью где то нашел ,про кривую память. Заменил в этом камне ,память на фирменную Винбонд. Сразу все заработало. Потом КДН не верил в такое. Пока ему не попался кривой чип. Он решил это прошивкой с параметрами QIO DIO. Че на че не помню и у него тоже кривой чип заработал , без замены памяти. Насчет этой темы ещё Andy в курсе.  А и при чем на Гайверовской простой лампе ,эти кривые чипы у меня ,работали ,без проблем подключения. Вроде ни че не забыл.

**Reply by dina16121974** [reply]
Я это к чему ,что память кривая может быть.

**Comment by vortigont** [comment]
В общем багов тут вагон и маленькая тележка. Думаю уходить на S3 чип может. Надо параллельно гонять.

**Reply by dina16121974** [reply]
У меня макетка ,готова на s3.

**Reply by dina16121974** [reply]
Если что ,я писал с 15 января в отпуске. Могу 3 макетки есп 32 запустить на круглосуточно. Давайте команду)))

**Comment by vortigont** [comment]
ды просто запускайте и оставьте как есть, у меня иногда через несколько часов отваливается, иногда сутками работает. У вас, я так понимаю вообще такого не было еще. Так что да, скорее это какое-то сочетание аппаратно/вайфайных проблем очень специфичных. Пока у меня идей новых нет.

**Reply by dina16121974** [reply]
Ок. Если что роутер у меня из китая Totolink 16 или 17 года.

**Comment by vortigont** [comment]
создал группу в телеге t.me/lampdevs
